### ISSUE TITLE
PLEASE add button to de-stretch any YouTube video while playing (in real time)

### ISSUE THREAD

**Issue by leodevbro** [TeamNewPipe/NewPipe#7805]
### Checklist

- [x] I checked, but didn't find any duplicates (open OR closed) of this issue in the repo. 
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x] This issue contains only one feature request. I will open one issue for every feature I want to request.

#### What feature do you want?
I want to have a button to change video aspect ratio any time.

#### Why do you want this feature?
Many videos were originally in 4:3 aspect ratio, but then uploaded on YouTube with 16:9 (stretched aspect ratio). For me it's very annoying to watch stretched videos. I tried the `v` button at the upper right, then tried the button `Fit/Fill/Zoom` but it does not change aspect ratio.

![a16-9](https://user-images.githubusercontent.com/54504880/152533565-e7a153c7-d75a-480c-96fd-bb5c09df3516.png)

**Comment by castrik** [comment]
I would also recommend here, a videi player called mx player, sadly its closed source, but its controls are the best on android, pinch hiorizontally/vertically to change dimensions, diagonally to zoom, swipe with two fingers up/down to change speed, and much more!
It would be great if we can implement those gesture features in newpipe's player

**Comment by SC1040-TS2** [comment]
> ### Checklist
> * [x]  I checked, but didn't find any duplicates (open OR closed) of this issue in the repo.
> * [x]  I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
> * [x]  This issue contains only one feature request. I will open one issue for every feature I want to request.
> 
> #### What feature do you want?
> I want to have a button to change video aspect ratio any time.
> 
> #### Why do you want this feature?
> Many videos were originally in 4:3 aspect ratio, but then uploaded on YouTube with 16:9 (stretched aspect ratio). For me it's very annoying to watch stretched videos. I tried the `v` button at the upper right, then tried the button `Fit/Fill/Zoom` but it does not correct the aspect ratio well enough.
> 
> ![a16-9](https://user-images.githubusercontent.com/54504880/152533565-e7a153c7-d75a-480c-96fd-bb5c09df3516.png)

Does this phenomenon occur even when the Playback is set to Fit, rather than Fill or Zoom?

Never mind it; it seems that using the Fit function failed to correct the video aspect ratio in that instance.

**Comment by leodevbro** [comment]
Yeah, the `Fit` does not do much in this case. The video is 1280x720 (16:9) resolution, but actually it's stretched, it should be 960x720 (4:3), so, we need to de-stretch it.

**Comment by Isssuemaker** [comment]
+1

**Comment by SameenAhnaf** [comment]
@Isssuemaker Please avoid making such comments. If you want to emphasize any issue, just upvote it.

**Comment by litetex** [comment]
I don't have much hop, that this will be implemented soon because the video was uploaded in a wrong format in the first place.

Anyway can we get a link of an example video (like that one in the screenshot)?
Might be helpful in the future.

**Comment by leodevbro** [comment]
I can provide entire playlists on YouTube.

For example **the images** I included above is from the most popular TV show in Georgia (Not USA) called "შუა ქალაქში" (2007) ["Shua Kalakshi"] ("In The Middle Of The City"). It is a Sitcom like "Friends" (1994).

This is the playlist of the series:
https://www.youtube.com/watch?v=dIkfBt4y4O8&list=PLcetRwIaiTnIFsDEUoS_WYvlZB8eT5NVe

As you can see there are **hundreds** of episodes with stretched (4:3 -> 16:9) aspect ratio. And I cannot find the episodes in original aspect ratio anywhere in YouTube or Google.

And I believe there are thousands of such stretched videos worldwide.


Currently my non-comfortable workaround is that I copy video link, then paste it in Chrome (Android), go to the video, then I paste this command in Chrome (Android) address bar:

`javascript:(function () { window.document.querySelector(".html5-video-container").style.transform = "scale(0.75, 1)"; })();`

And the video gets de-stretched. But Android Chrome browser is very clunky software to watch video with full comfort.

**Comment by SameenAhnaf** [comment]
Are you talking about https://github.com/TeamNewPipe/NewPipe/issues/7179#issuecomment-938116163??

**Comment by leodevbro** [comment]
Ah, yeah, I guess this issue is a duplicate of #7179, but I'll just add one note: manual de-stretching is good and gives us much freedom, but it will be better to have also some presets, the most important preset I guess would be the 16:9 -> 4:3 de-stretch preset, because I think most of the stretched videos are 4:3 -> 16:9 stretched.
